atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -206,24 +212,25 @@ private synchronized void updateStores(TransactionsLockWatchUpdate updateForTran
         Multimap<Sequence, StartTimestamp> reversedMap = createSequenceTimestampMultimap(updateForTransactions);
 
         // Without this block, updates with no events would not store a snapshot.
-        currentVersion.map(LockWatchVersion::version).map(Sequence::of).ifPresent(sequence -> {
+        Optional<Sequence> maybeSeq =
+                currentVersion.map(LockWatchVersion::version).map(Sequence::of);
+        if (maybeSeq.isPresent()) {
             synchronized (this) {
-                snapshotStore.storeSnapshot(sequence, reversedMap.get(sequence), valueStore.getSnapshot());
+                snapshotStore.storeSnapshot(maybeSeq.get(), reversedMap.get(maybeSeq.get()), valueStore.getSnapshot());
             }
-        });
+        }
 
-        updateForTransactions.events().stream().filter(this::isNewEvent).forEach(event -> {
-            synchronized (this) {
+        for (LockWatchEvent event : updateForTransactions.events()) {
+            if (isNewEvent(event)) {
```

Comment:
presumably did this to make the check happy without the need to synchro every time

Reply:
yeah, just unwound the streams to avoid synchronized in lambdas to make compiler happy